framework-generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/EventSourcesDestinationTypeFinder.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.raml.jms.core;
+
+import static java.util.Optional.empty;
+
+import uk.gov.justice.services.generators.subscription.parser.EventSourcesFileParser;
+import uk.gov.justice.services.generators.subscription.parser.EventSourcesFileParserFactory;
+import uk.gov.justice.services.generators.subscription.parser.JmsUriToDestinationConverter;
+import uk.gov.justice.subscription.domain.eventsource.EventSourceDefinition;
+import uk.gov.justice.subscription.jms.core.JmsUriToDestinationTypeConverter;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.jms.Destination;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * In the past, when processing raml files for generating MDBs, the envent-sources.yaml files where jms_uri are defined were ignored.
+ * This class is allowing to read any event-sources.yaml files in the classpath and trying to find the corresponding jms_uri
+ * that is used to determine whether we are dealing with a topic or a queue.
+ * Note that this is only used for event-processors only
+ */
+public class EventSourcesDestinationTypeFinder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventSourcesDestinationTypeFinder.class);
+
+    private final JmsUriToDestinationTypeConverter uriToDestinationTypeConverter = new JmsUriToDestinationTypeConverter();
+    JmsUriToDestinationConverter jmsUriToDestinationConverter = new JmsUriToDestinationConverter();
```

Comment:
Should JmsUriToDestinationConverter be private and final like the others?

Reply:
Fixed.
 I was just testing your vigilence :smile: @allanmckenzie